hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/BlocklistParser.java

```diff
@@ -0,0 +1,145 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl;
+
+import com.hedera.node.app.service.evm.utils.EthSigsUtils;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.HexFormat;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.asn1.sec.SECNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+
+/**
+ * Encapsulates the logic for reading blocked accounts from file
+ */
+public class BlocklistParser {
+    private static final Logger log = LogManager.getLogger(BlocklistParser.class);
+
+    /**
+     * todo
+     * Makes sure that all blocked accounts contained in the blocklist resource are present in state, and creates their definitions (if necessary).
+     *
+     * <p><b>Note: this method assumes that blocklists are enabled</b> – it does not check that config property
+     */
+    public List<BlockedInfo> parse(@NonNull final String blocklistResourceName) {
+        final List<String> fileLines = readFileLines(blocklistResourceName);
+        if (fileLines.isEmpty()) return Collections.emptyList();
+
+        return parseBlockList(fileLines);
+    }
+
+    private List<String> readFileLines(@NonNull final String blocklistResourceName) {
+        final List<String> fileLines;
+        try {
+            fileLines = readPrivateKeyBlocklist(blocklistResourceName);
+        } catch (Exception e) {
+            log.error("Failed to read blocklist resource {}", blocklistResourceName, e);
+            return Collections.emptyList();
+        }
+        return fileLines;
+    }
+
+    private static List<BlockedInfo> parseBlockList(final List<String> fileLines) {
+        final List<BlockedInfo> blocklist;
+        try {
+            final var columnHeaderLine = fileLines.get(0); // Assume that the first line is the header
+            final var blocklistLines = fileLines.subList(1, fileLines.size());
+            final var columnCount = columnHeaderLine.split(",").length;
+            blocklist = blocklistLines.stream()
+                    .map(line -> parseCSVLine(line, columnCount))
+                    .toList();
+        } catch (IllegalArgumentException iae) {
+            log.error("Failed to parse blocklist", iae);
+            return Collections.emptyList();
+        }
+        return blocklist;
+    }
+
+    @NonNull
+    private static List<String> readPrivateKeyBlocklist(@NonNull final String fileName) {
+        final var inputStream = BlocklistParser.class.getClassLoader().getResourceAsStream(fileName);
```

Comment:
You are not closing the streams. Maybe you can put everything in a try-with-resources statement.

Reply:
Done

Reply:
You also have to close `inputStream`. Luckily, `try-with-resource` provides support for multiple resources.

Reply:
Good catch. Done